src/dlgTriggerEditor.cpp

```diff
@@ -2842,21 +2777,23 @@ void dlgTriggerEditor::slot_timer_toggle_active()
             }
         }
     }
+
     if (pT->state()) {
-        pItem->setIcon(0, icon);
-        pItem->setText(0, pT->getName());
+        if (pT->shouldBeActive()) {
+            showInfo(tr(R"(Trying to activate a timer group, offset timer, timer or the part of a module "%1" that contains them <em>succeeded</em>.)").arg(pT->getName()));
+        } else {
+            showInfo(tr(R"(Trying to deactivate a timer group, offset timer, timer or the part of a module "%1" that contains them <em>succeeded</em>.)").arg(pT->getName()));
+        }
     } else {
-        QIcon iconError;
-        iconError.addPixmap(QPixmap(QStringLiteral(":/icons/tools-report-bug.png")), QIcon::Normal, QIcon::Off);
-        pItem->setIcon(0, iconError);
+        pT->setIsActive(false);
+        showError(tr(R"(Unable to activate (and automatically deactivating) an offset timer or timer or the part of a module "%1" that contains them; reason: %2.)").arg(pT->getName(), pT->getError()));
```

Comment:
> Since we're unable to activate it, it means that it's only other possible state is that it is deactivated.

Not exactly - it is, what I have previously started to use: *bugged-out* - i.e. in an error state - which, whilst it effectively the same as disabled in that it does not do anything - means it cannot be activated without user action however many times they click `activate`.

The text in this area, *importantly* also covers the case which we haven't really dwelt on before - of a ***module*** - where the brown folder is merely a representation of (in this case, the timer type) elements of that module - the significance of that is **it is a dummy *and cannot be edited!*** It IS possible to disable all of that type of elements in the module from the editor - but anything else is not effective - anything the user enters in any of the fields is forgotten when they switch to another item (and changing the name is not possible because it is set from the module name and also cannot be changed - and given that there is likely to be identical folders in other views that is just as well,..!)

Reply:
Also, the code previous to this commit would report that (in the case of said bugging-out) it was not possible to **deactivate** the item for whatever reason - which is confusingly rong! This text also handles the situation where it *was* activate but the user has since saved some new, bugged code - this PR means that it now forces the item to be deactivated (and is saying so) - and they will need to reactivate it once any bugs have been fixed.  This might sound a bit less friendly but it is also less confusing - if it is broken the user must switch it back on after it is fixed - of course the double-click on the item in the treewidget will do both the save (and compile - which will clear the error state) and then the activation...

Reply:
Technically true, but practically confusing. I'd still opt for the simpler version.